src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/SourceRepositoryExtensions.cs

```diff
@@ -166,17 +169,20 @@ public static Task<SearchResult<IPackageSearchMetadata>> SearchAsync(this Source
                 // filter packages based on allowed versions
                 var updatedPackages = packages.Where(p => allowedVersions.Satisfies(p.Identity.Version));
 
+                cancellationToken.ThrowIfCancellationRequested();
```

Comment:
No work has been done at this point since `metadataResource.GetMetadataAsync` was called. (LINQ uses deferred execution)